server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -244,13 +277,15 @@ Map<Long, ThreadTimeAccumulator> getAllValidThreadInfos(ThreadMXBean threadBean,
 
     ThreadInfo[][] captureThreadStacks(ThreadMXBean threadBean, long[] threadIds) throws InterruptedException {
         ThreadInfo[][] result = new ThreadInfo[threadElementsSnapshotCount][];
-        for (int j = 0; j < threadElementsSnapshotCount; j++) {
-            // NOTE, javadoc of getThreadInfo says: If a thread of the given ID is not alive or does not exist,
-            // null will be set in the corresponding element in the returned array. A thread is alive if it has
-            // been started and has not yet died.
+
+        // NOTE, javadoc of getThreadInfo says: If a thread of the given ID is not alive or does not exist,
+        // null will be set in the corresponding element in the returned array. A thread is alive if it has
+        // been started and has not yet died.
+        for (int j = 0; j < threadElementsSnapshotCount - 1; j++) {
             result[j] = threadBean.getThreadInfo(threadIds, Integer.MAX_VALUE);
             Thread.sleep(threadElementsSnapshotDelay.millis());
         }
+        result[threadElementsSnapshotCount - 1] = threadBean.getThreadInfo(threadIds, Integer.MAX_VALUE);
```

Comment:
This avoids sleep after collecting the last sample.
Should make collection much faster when executing with `threadElementsSnapshotCount=1`

Reply:
Good idea! Wrapping the sleep inside a `if (j < threadElementsSnapshotCount - 1)` check might be a smaller changeset.

Reply:
nit: maybe instead adding an if in the for loop is a bit more readable.

Reply:
I prefer to make a loop one iteration shorter rather than every time check if it is the last iteration.